public/app/features/integrated-alerting/components/NotificationChannel/AddNotificationChannelModal/AddNotificationChannelModal.tsx

```diff
@@ -37,18 +40,36 @@ export const AddNotificationChannelModal: FC<AddNotificationChannelModalProps> =
       logger.error(e);
     }
   };
-  const renderTypeFields = useCallback((values: NotificationChannelRenderProps) => {
-    const TypeFields = TYPE_FIELDS_COMPONENT[values.type.value];
+  const renderTypeFields = useCallback(
+    (values: NotificationChannelRenderProps, mutators: Record<string, (...args: any[]) => any>) => {
+      const TypeFields = TYPE_FIELDS_COMPONENT[values.type.value];
 
-    return <TypeFields values={values} />;
-  }, []);
+      // By passing down the mutators, these fields will be able to call them
+      return <TypeFields values={values} mutators={mutators} />;
+    },
+    []
+  );
```

Comment:
usually the `render*` functions in React components are a code smell, since they can often be converted to component themselves. I wonder if we can do the same here

Reply:
Sure. I just added the mutators, the function has already there 😅 
But I agree with you. I've changed it to a component ;)